### PR TITLE
Fixed log constants

### PR DESCRIPTION
**Pull request by andreybolonin**
```sh
PHP Notice 'yii\base\ErrorException' with message 'Use of undefined constant NET_SFTP_LOG_REALTIME - assumed 'NET_SFTP_LOG_REALTIME''

in /data/cporder-2-0/vendor/phpseclib/phpseclib/phpseclib/Net/SFTP.php:2682

Stack trace:
#0 /data/cporder-2-0/vendor/phpseclib/phpseclib/phpseclib/Net/SFTP.php(2682): yii\base\ErrorHandler->handleError(8, 'Use of undefine...', '/data/cporder-2...', 2682, Array)
#1 /data/cporder-2-0/vendor/phpseclib/phpseclib/phpseclib/Net/SFTP.php(467): phpseclib\Net\SFTP->_send_sftp_packet(1, '\x00\x00\x00\x03')
#2 /data/cporder-2-0/components/Transports/TransportSFTP.php(29): phpseclib\Net\SFTP->login('CentralPoint', 'VoOe8TVK4QW5mB9...')
#3 /data/cporder-2-0/components/Transports/TransportBase.php(55): app\components\Transports\TransportSFTP->getChannel()
#4 /data/cporder-2-0/commands/ServiceController.php(741): app\components\Transports\TransportBase->__construct(Array)
#5 [internal function]: app\commands\ServiceController->actionTie()
#6 /data/cporder-2-0/vendor/yiisoft/yii2/base/InlineAction.php(55): call_user_func_array(Array, Array)
#7 /data/cporder-2-0/vendor/yiisoft/yii2/base/Controller.php(151): yii\base\InlineAction->runWithParams(Array)
#8 /data/cporder-2-0/vendor/yiisoft/yii2/console/Controller.php(91): yii\base\Controller->runAction('tie', Array)
#9 /data/cporder-2-0/vendor/yiisoft/yii2/base/Module.php(455): yii\console\Controller->runAction('tie', Array)
#10 /data/cporder-2-0/vendor/yiisoft/yii2/console/Application.php(167): yii\base\Module->runAction('service/tie', Array)
#11 /data/cporder-2-0/vendor/yiisoft/yii2/console/Application.php(143): yii\console\Application->runAction('service/tie', Array)
#12 /data/cporder-2-0/vendor/yiisoft/yii2/base/Application.php(375): yii\console\Application->handleRequest(Object(yii\console\Request))
#13 /data/cporder-2-0/yii(15): yii\base\Application->run()
#14 {main}
```

```sh
PHP Notice 'yii\base\ErrorException' with message 'ob_flush(): failed to flush buffer. No buffer to flush'

in /data/cporder-2-0/vendor/phpseclib/phpseclib/phpseclib/Net/SFTP.php:2757

Stack trace:
#0 [internal function]: yii\base\ErrorHandler->handleError(8, 'ob_flush(): fai...', '/data/cporder-2...', 2757, Array)
#1 /data/cporder-2-0/vendor/phpseclib/phpseclib/phpseclib/Net/SFTP.php(2757): ob_flush()
#2 /data/cporder-2-0/vendor/phpseclib/phpseclib/phpseclib/Net/SFTP.php(472): phpseclib\Net\SFTP->_get_sftp_packet()
#3 /data/cporder-2-0/components/Transports/TransportSFTP.php(29): phpseclib\Net\SFTP->login('CentralPoint', 'VoOe8TVK4QW5mB9...')
#4 /data/cporder-2-0/components/Transports/TransportBase.php(55): app\components\Transports\TransportSFTP->getChannel()
#5 /data/cporder-2-0/commands/ServiceController.php(741): app\components\Transports\TransportBase->__construct(Array)
#6 [internal function]: app\commands\ServiceController->actionTie()
#7 /data/cporder-2-0/vendor/yiisoft/yii2/base/InlineAction.php(55): call_user_func_array(Array, Array)
#8 /data/cporder-2-0/vendor/yiisoft/yii2/base/Controller.php(151): yii\base\InlineAction->runWithParams(Array)
#9 /data/cporder-2-0/vendor/yiisoft/yii2/console/Controller.php(91): yii\base\Controller->runAction('tie', Array)
#10 /data/cporder-2-0/vendor/yiisoft/yii2/base/Module.php(455): yii\console\Controller->runAction('tie', Array)
#11 /data/cporder-2-0/vendor/yiisoft/yii2/console/Application.php(167): yii\base\Module->runAction('service/tie', Array)
#12 /data/cporder-2-0/vendor/yiisoft/yii2/console/Application.php(143): yii\console\Application->runAction('service/tie', Array)
#13 /data/cporder-2-0/vendor/yiisoft/yii2/base/Application.php(375): yii\console\Application->handleRequest(Object(yii\console\Request))
#14 /data/cporder-2-0/yii(15): yii\base\Application->run()
#15 {main}
```

### CODE REVIEW

**Comment by terrafrost** [comment]
Nice find.

This should be merged into the 2.0 branch, however, and not into master. Once it's merged into the 2.0 branch the 2.0 branch can be merged into master.

Also, a unit test would be nice. I wouldn't say it's a requirement but it'd be nice :)

**Comment by andreybolonin** [comment]
Fixed
```sh
@ob_flush()
````

What about https://github.com/FriendsOfPHP/PHP-CS-Fixer code style fixes ?

**Comment by bantu** [comment]
Please keep this PR to bugfix only.

**Comment by andreybolonin** [comment]
All is ok?

**Comment by terrafrost** [comment]
Can you squash the commits? Like right now you modified a whole bunch of unrelated lines and then undid those changes. But since the commits are still there git blame will still show that they were changed and merge conflicts will still be able to occur on those lines. Squashing the commits should essentially undo the changes :)

File: phpseclib/Net/SFTP.php [hunk 1]
```diff
@@ -2752,13 +2751,13 @@ function _get_sftp_packet()
         if (defined('NET_SFTP_LOGGING')) {
             $packet_type = '<- ' . $this->packet_types[$this->packet_type] .
                            ' (' . round($stop - $start, 4) . 's)';
-            if (NET_SFTP_LOGGING == NET_SFTP_LOG_REALTIME) {
+            if (NET_SFTP_LOGGING == self::LOG_REALTIME) {
                 echo "<pre>\r\n" . $this->_format_log(array($packet), array($packet_type)) . "\r\n</pre>\r\n";
                 flush();
-                ob_flush();
+                @ob_flush();
```
Comment (bantu) [inline phpseclib/Net/SFTP.php]: Unrelated change

File: phpseclib/Net/SFTP.php [hunk 2]
```diff
@@ -2676,13 +2675,13 @@ function _send_sftp_packet($type, $data)
         if (defined('NET_SFTP_LOGGING')) {
             $packet_type = '-> ' . $this->packet_types[$type] .
                            ' (' . round($stop - $start, 4) . 's)';
-            if (NET_SFTP_LOGGING == NET_SFTP_LOG_REALTIME) {
+            if (NET_SFTP_LOGGING == self::LOG_REALTIME) {
                 echo "<pre>\r\n" . $this->_format_log(array($data), array($packet_type)) . "\r\n</pre>\r\n";
                 flush();
-                ob_flush();
+                @ob_flush();
```
Comment (bantu) [inline phpseclib/Net/SFTP.php]: Unrelated change

**Comment by terrafrost** [comment]
I've just made my own commit of this:

https://github.com/phpseclib/phpseclib/commit/b3171cc63f44c715a2b29200d9549d866e94d123

File: phpseclib/Net/SSH2.php [hunk 3]
```diff
@@ -3482,7 +3482,7 @@ function _append_log($message_number, $message)
             case self::LOG_REALTIME_FILE:
                 if (!isset($this->realtime_log_file)) {
                     // PHP doesn't seem to like using constants in fopen()
-                    $filename = self::LOG_REALTIME_FILENAME;
+                    $filename = self::LOG_REALTIME_FILE;
```
Comment (terrafrost) [inline phpseclib/Net/SSH2.php]: This should never have been made a constant in the first place - rather, it should have been NET_SFTP_LOGGING_REALTIME_FILE. The reason being so that you can define the name yourself.

But overall I don't like the whole logging system anyway and at some point in the master branch I intend to redo the whole thing.
